### ISSUE TITLE
Privacy concerns

### ISSUE THREAD

**Issue by RobinIsTheBird** [FDPA/resolutions#2]
This repository is public, and is deployed via github.io, also public.

The html has google API keys embedded in it. The google developers console warns not to do this.

As long as resolutions are hosted by github.io, the API keys must be public. Are the public API keys enough of a concern to move the resolutions page and its embeddable counterpart to a hosting service, and transmitting the keys in a cookie over https?

Looking for weigh in by @amyprime, @sarawb, and @wnstn on whether this is a problem.

**Comment by amyprime** [comment]
Since the API is associated with my account, and I haven't given Google any authorization to charge me money, I think the worst thing that can happen is that it gets used up and we can't see the map for 24 hours.  Basically, I'm willing to close this as "not important at this time."

**Comment by RobinIsTheBird** [comment]
Yeah, the most likely attack on an API key is denial of service. I suppose there might be some way to use an API key to break into the owner's Google account, or to add a spam or phishing sniffer to the fusiontables. I own another API key to those tables for dev purposes, also embedded in the code. Mine is restricted to IP addresses that are strictly for localhost use, but if the API key opens a door for the attacker to change the key's restrictions, my key is a similar risk.

The Google fusiontables documentation just said not to make API keys public. I'm guessing they didn't elaborate so as not to give would-be attackers any ideas.

My 2 cents is this:
1. Any privacy risk either to Amy and me or to the targets of spam & phishing is not really acceptable, and putting the code in the FDPA github repo just made it more accessible to prying eyes.
2. If it was re-created in the fairdistrictspa.com domain, then we could build a Craft template for it directly rather than iframing it.
3. As a Craft template, the API keys could be added to the deployment environment and kept out of the code. As a side benefit, we could eliminate all the `isProduction` logic in the `JavaScript`.
4. We could point the github.io domains at a `index.html` with a forwarding meta tag to forward to the new page, so people using the old URL wouldn't have to update it.

**Comment by amyprime** [comment]
I'm officially out of my domain now in this thread, but it sounds like this might be a good way for @RobinIsTheBird to get your feet wet with Craft.

**Comment by amyprime** [comment]
It just occurred to me that I could make a new API key under the account that will own the tables.  Then at least it won't be tied to me personally.

**Comment by RobinIsTheBird** [comment]
Two API keys, one for production, one for development.

Still shouldn't be embedded in the source code.

Happy to do the dev to add this to Craft, but I don't know who authorizes work on the website. Sara? Also, @wnstn has been assigned to iframe the maps into the Craft resolutions page. Do we want to do that in order to get something out quickly until I'm ready to make a template, or is iframe low enough priority from a user point of view to wait until we have security/privacy worked out?

**Comment by amyprime** [comment]
@RobinIsTheBird I think you can just go ahead and delete the development API key from the code, since it's entirely transparent to any end user.  Similarly, if I set up a different account/key, I'm just going to go ahead and swap out the production one.

**Comment by amyprime** [comment]
According to Amy R., getting the map onto the FDPA website is indeed a high priority, so my vote would be to iframe it ASAP, and then work on embedding it into Craft, with associated security benefits.  I don't think the iframing would be any worse security-wise than what we have now -- it's already linked from the main site, so crawlers will find it regardless, and the amyprime.github.io version was already shared publicly for a long time before that.

@wnstn @sarawb Do you have any issues if @RobinIsTheBird wants to go ahead and work the iframing task, since it sounds like she's available to do so now?

**Comment by sarawb** [comment]
I don't have an issue with that, but I know @wnstn has a lot of experience with iframing in content accessibly and to align with the rest of the experience, so I would look to him to weigh in on approach here.

**Comment by RobinIsTheBird** [comment]
I have deleted my development API key. The code in this repo still refers to it.

New development keys can be created using the new fdpa gmail account whatever we end up calling it.

@wnstn will be able to do the `iframing` quicker than I can, since I still need to go through onboarding. If we can create an issue on `FDPA/fdpa` for the Craft template and assign it to me, then I'll happily complete onboarding and get to work on it.

In one sense, the security isn't worse than it was under `amyprime.github.io`, but `fairdistrictspa.com` is much easier to find, and with the `iframe`, the new github.io site will be easier to find, and a malicious snooper could figure all sorts of things out from there. Sort of like the difference between publishing private information in the Swarthmorean vs. the Philadelphia Inquirer. Makes me nervous that someone might exploit the information before we have a chance to withdraw it.

Once templated in Craft, we should change the key again, just in case someone got it earlier when it was public.
